docs/JAX_basic_text_classification.md

```diff
@@ -0,0 +1,478 @@
+            pbar.set_postfix({"loss": loss.item()})
+            pbar.update(1)
+
+
+def evaluate_model(epoch):
+    # Compute the metrics on the train and val sets after each training epoch.
+    model.eval()  # Set model to evaluation model: e.g. use stored batch statistics
+
+    eval_metrics.reset()  # Reset the eval metrics
+    for test_batch in test_loader:
+        eval_step(model, test_batch, eval_metrics)
+
+    for metric, value in eval_metrics.compute().items():
+        eval_metrics_history[f'test_{metric}'].append(value)
+
+    print(f"[test] epoch: {epoch + 1}/{num_epochs}")
+    print(f"- total loss: {eval_metrics_history['test_loss'][-1]:0.4f}")
+    print(f"- Accuracy: {eval_metrics_history['test_accuracy'][-1]:0.4f}")
+```
+
+Now, we can start the training.
+
+```{code-cell} ipython3
+%%time
+
+for epoch in range(num_epochs):
+    train_one_epoch(epoch)
+    evaluate_model(epoch)
+```
+
+Let's visualize the collected metrics:
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+
+
+plt.plot(train_metrics_history["train_loss"], label="Loss value during the training")
+plt.legend()
+```
+
+```{code-cell} ipython3
+fig, axs = plt.subplots(1, 2, figsize=(10, 10))
+axs[0].set_title("Loss value on test set")
+axs[0].plot(eval_metrics_history["test_loss"])
+axs[1].set_title("Accuracy on test set")
+axs[1].plot(eval_metrics_history["test_accuracy"])
+```
+
+We can observe that the model starts overfitting after the 5-th epoch and the best accuracy it could achieve is around 0.87. Let us also check few model's predictions on the test data:
+
+```{code-cell} ipython3
+data = test_dataset[10]
+```
+
+```{code-cell} ipython3
+text_processing = TextPreprocessing(tokenizer, max_length=max_length)
+processed_data = text_processing.map(data)
+model.eval()
+preds = model(processed_data["text"][None, :])
+pred_label = preds.argmax(axis=-1).item()
+confidence = nnx.softmax(preds, axis=-1)
+
+print("- Text:\n", data["text"])
+print("")
+print(f"- Expected review sentiment: {'positive' if data['label'] == 0 else 'negative'}")
+print(f"- Predicted review sentiment: {'positive' if pred_label == 0 else 'negative'}, confidence: {confidence[0, pred_label]:.3f}")
+```
+
+```{code-cell} ipython3
+data = test_dataset[20]
+```
+
+```{code-cell} ipython3
+text_processing = TextPreprocessing(tokenizer, max_length=max_length)
+processed_data = text_processing.map(data)
+model.eval()
+preds = model(processed_data["text"][None, :])
+pred_label = preds.argmax(axis=-1).item()
+confidence = nnx.softmax(preds, axis=-1)
+
+print("- Text:\n", data["text"])
+print("")
+print(f"- Expected review sentiment: {'positive' if data['label'] == 0 else 'negative'}")
+print(f"- Predicted review sentiment: {'positive' if pred_label == 0 else 'negative'}, confidence: {confidence[0, pred_label]:.3f}")
+```
+
+## Further reading
+
+- Model checkpointing and exporting using [Orbax](https://orbax.readthedocs.io/en/latest/)
```

Comment:
It would be nice to have some sort of conclusion text here – summarize what we did and what the next steps may be for someone interested in going deeper.

Reply:
Yes, I wanted to add links on other NLP tutorials doing text classification with transformers once we have them.
I'll add a short summary on what we did.